base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java

```diff
@@ -550,7 +551,13 @@ private void retrieveCapability(final SystemCapabilityType systemCapabilityType,
 		}
 		final GetSystemCapability request = new GetSystemCapability();
 		request.setSystemCapabilityType(systemCapabilityType);
-		// If subscribe is null, then don't change the current subscription status
+
+		/*
+		The subscription flag in the request should be set based on multiple variables:
+		- if subscribe is null (no change), shouldSubscribe = current subscription status
+		- if subscribe is false, then shouldSubscribe = false
+		- if subscribe is true and the HU supports subscriptions, then shouldSubscribe = true
```

Comment:
But if the HU doesn't support subscriptions, `shouldSubscribe` will be `false` right? even if `subscribe = true`? because we are doing && supportsSubscriptions()

Reply:
Based on the current code, I don't think that's true.

```java 
if (subscribe != null) {
    return subscribe;
} else {
    return isSubscribedToSystemCapability(systemCapabilityType)) && supportsSubscriptions();
}
```

So if `subscribe == true`, then it will always return `true`.

Reply:
It is actually confusing. I will simplify the condition a bit.